File: algorithms/27.py
# https://leetcode.com/problems/remove-element


class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        if nums is None:
            return 0

        i = 0
        while i < len(nums):
            if nums[i] == val:
                del nums[i]
            else:
                i += 1

File: algorithms/27.swift
// https://leetcode.com/problems/remove-element

class Solution {
    func removeElement(inout nums: [Int], _ val: Int) -> Int {
        var i = 0
        while i < nums.count {
            if nums[i] == val { nums.removeAtIndex(i) }
            else { i += 1 }
        }
        return nums.count
    }
}
